DigitalLearningSolutions.Data.Tests/DataServices/UserDataServiceTests/UserCentreDetailsDataServiceTests.cs

```diff
@@ -117,5 +117,43 @@ public void GetUnverifiedCentreEmailsForUser_returns_unverified_emails_for_activ
             result.Should().HaveCount(1);
             result[0].centreEmail.Should().BeEquivalentTo("unverified@centre.email");
         }
+
+        [Test]
+        [TestCase("test@gmail.com", true)]
+        [TestCase("not_an_email_in_the_database", false)]
+        public void PrimaryEmailIsInUse_returns_expected_value(string email, bool expectedResult)
+        {
+            // When
+            var result = userDataService.PrimaryEmailIsInUse(email);
+
+            // Then
+            result.Should().Be(expectedResult);
+        }
+
+        [Test]
+        [TestCase("centre@email.com", true)]
+        [TestCase("not_an_email_in_the_database", false)]
+        public void CentreSpecificEmailIsInUseAtCentre_returns_expected_value(string email, bool expectedResult)

```

Comment:
Is it worth testing this when the email exists, but at a different centre?

Reply:
It is indeed